schema_based_on_diagram.sql

```diff
@@ -0,0 +1,51 @@
+CREATE TABLE patients(
+   id serial PRIMARY KEY,
+   name VARCHAR(255) NOT NULL,
+   date_of_birth DATE
+);
+
+CREATE TABLE treatments(
+   id serial PRIMARY KEY,
+   type VARCHAR(100),
+   name VARCHAR(250)
+);
+
+CREATE TABLE medical_histories(
```

Comment:
- [ ] Kindly make sure the `many-to-many` relationship between the medical_history and treatments is creating through an intermediate table.
Hint: You may consider saying:
```sql
CREATE TABLE medical_histories_has_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
    );
```

Reply:
- [ ] Also make sure all your foreign keys have their indexes as specified in the project requirements.
e.g:
```sql
CREATE INDEX ON medical_histories (patient_id);
-- and so on --
```